tracing-appender/src/rolling.rs

```diff
@@ -500,13 +507,40 @@ impl Inner {
                     .unwrap_or(0),
             ),
             rotation,
+            keep_last: None,
         };
         (inner, writer)
     }
 
     fn refresh_writer(&self, now: OffsetDateTime, file: &mut File) {
         let filename = self.rotation.join_date(&self.log_filename_prefix, &now);
 
+        if let Some(keep_last) = self.keep_last {
+            let mut files = vec![];
+            for file in fs::read_dir(&self.log_directory).unwrap() {
+                let path = file.unwrap().path();
+                let filename = path.clone().into_os_string().into_string().unwrap();
+                if filename.contains(&self.log_filename_prefix) {
+                    files.push(path);
+                }
+            }
+            if files.len() > (keep_last - 1) {
+                let mut sorted_files = vec![];
+                for file in &files {
+                    let filename = file.clone().into_os_string().into_string().unwrap();
+                    let date_str =
+                        filename.clone()[(filename.rfind('.').unwrap() + 1)..].to_string();
+                    let date: usize = date_str.replace("-", "").parse().unwrap();
```

Comment:
It might not work if the underlying format of date would be changed.

I would extract `Rotation::join_date` into a type which would implement both `Display` and `FromStr`